rclcpp/include/rclcpp/node.hpp

```diff
@@ -933,6 +938,11 @@ class Node : public std::enable_shared_from_this<Node>
   OnSetParametersCallbackHandle::SharedPtr
   add_on_set_parameters_callback(OnParametersSetCallbackType callback);
 
+  RCLCPP_PUBLIC
+      RCUTILS_WARN_UNUSED
+  OnSetLocalParametersCallbackHandle::SharedPtr
+  add_local_parameters_callback(OnLocalParametersSetCallbackType callback);
```

Comment:
Similar t the nomenclature of ``add_n_set_parameters_callback``, we should have this as ``add_post_set_parameters_callback``

Reply:
``OnSetLocalParametersCallbackHandle`` can be changed to ``PostSetParametersCallbackHandle``

Reply:
IMO, we should consider moving away from the current naming scheme. I think "on_set_parameters" is confusing, and should probably be deprecated. I also think the term "local" is kind of redundant since both callbacks in question deal with parameters local to the node. It's only when a user uses ParameterEventsHandler (which is separate from the node) that they can get parameter events for other nodes.

Changing the name of the existing functions will result in more churn, but I think it's worth taking the time now to reflect on what we want the API to look like.

It's just my opinion, but I would ultimately like to see:

- `add_parameters_validation_callback`, as the new name for the validation feature.
- `add_parameters_set_callback` (or `add_on_parameters_set_callback`), as the name for the new feature.

Reply:
> It's just my opinion, but I would ultimately like to see:
>
>    add_parameters_validation_callback, as the new name for the validation feature.
>    add_parameters_set_callback (or add_on_parameters_set_callback), as the name for the new feature.

If I were designing both of these APIs from the ground-up, I would agree with you.  But I'm actually somewhat reluctant to make this API change now because:

1. We already renamed the existing API twice before.  For Eloquent, we renamed the original `register_param_change_callback` to `set_on_parameters_set_callback`, and then in Galactic we changed it to `add_on_set_parameters_callback`.  While we can keep renaming it, we incur costs on the downstream consumers every time we do it.
2. `add_on_parameters_set_callback` is too similar to `add_on_set_parameters_callback`, even though they have different semantics.

I'm not totally sure what to do about this, but I thought I would point it out.  At the very least I think the second one should have a different name.